src/coreir-c/coreir-c.cpp

```diff
@@ -455,6 +455,15 @@ const char* COREModuleGetMetaData(COREModule* core_mod) {
   return cstr;
 }
 
+const char* COREModuleGetMetaData(COREGenerator* core_gen) {
+  Generator* gen = rcast<Generator*>(core_gen);
+  string mstr = gen->getMetaData().dump();
+  std::size_t len = mstr.size() + 1;
+  char* cstr = (char*)malloc(len);
```

Comment:
How does this get freed? Looks like the instance metadata API allocates from context's scratchmemory.